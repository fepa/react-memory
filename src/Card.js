import React from 'react';

class Card extends React.Component {
  isMatched() {
    return this.props.matchedPairs[this.props.pairing];
  }

  isActive() {
    return this.props.activeCards.indexOf(this) > -1;
  }

  isFaceUp() {
    return this.isMatched() || this.isActive();
  }

  matches(card) {
    return this.props.pairing === card.props.pairing;
  }

  clickHandler() {
    if(this.isMatched() || this.isActive()) {
      return;
    } else {
      this.props.checkMatch(this);
    }
  }

  render() {
    return (
      <div className={"card " + (this.isFaceUp() ? '' : '-face-down')} onClick={this.clickHandler.bind(this)}>
        <img src={this.props.imageURL} alt="Card" />
      </div>
    );
  }
};

export default Card;
