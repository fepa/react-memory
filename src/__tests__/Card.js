import React from 'react';
import ReactDOM from 'react-dom';
import Card from '../Card';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  const instance = <Card matchedPairs={{}} activeCards={[]} imageURL='foo' />;
  ReactDOM.render(instance, div);
});

it('renders face up when matched', () => {
  const component = renderer.create(
    <Card pairing="1" matchedPairs={{"1": true}} activeCards={[]} imageURL='foo' />
  );
  expect(component.toJSON()).toMatchSnapshot();
});

// it('renders face up when active', () => {
//   console.log('active');
//   const inst = <Card pairing="1" matchedPairs={{}} activeCards={[]} />;
//   inst.props.activeCards.push(inst);
//   const component = renderer.create(
//     inst,
//   );

//   let tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
// });

it('renders face down when not matched and not active', () => {
  const component = renderer.create(
    <Card pairing="1" matchedPairs={{}} activeCards={[]} imageURL='foo' />,
  );
  expect(component.toJSON()).toMatchSnapshot();
});
