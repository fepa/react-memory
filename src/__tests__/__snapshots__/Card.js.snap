// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders face down when not matched and not active 1`] = `
<div
  className="card -face-down"
  onClick={[Function]}
>
  <img
    alt="Card"
    src="foo"
  />
</div>
`;

exports[`renders face up when matched 1`] = `
<div
  className="card "
  onClick={[Function]}
>
  <img
    alt="Card"
    src="foo"
  />
</div>
`;
